### PR TITLE
Added multiple snippets

### PR DESCRIPTION
**Pull request by anderseide**
Added snippets for NSG, NSG Rule, ParametersFile, Deployment Parameter, MetadataFile

### CODE REVIEW

File: VSCode/armsnippets.json [hunk 1]
```diff
@@ -15,6 +15,30 @@
     ],
     "description": "Skeleton ARM Template"
 },
+"Azure Resource Manager (ARM) Parameters Template": {
```
Comment (philon-msft) [inline VSCode/armsnippets.json]: I'd rather not introduce a snippet for parameters. It should just be a single property ("parameters": {}), and the template snippet above already provides that.

File: VSCode/armsnippets.json [hunk 2]
```diff
@@ -15,6 +15,30 @@
...
+    "body": [
+        "{",
+        "    \"$$schema\": \"https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#\",",
+        "    \"contentVersion\": \"1.0.0.0\",",
+        "    \"parameters\": {}",
+        "}"
+    ],
+    "description": "Skeleton ARM Parameters Template"
+},
+"Azure Resource Manager (ARM) Metadata Template": {
```
Comment (philon-msft) [inline VSCode/armsnippets.json]: This metadata isn't specific to ARM templates, so I'd rather not add it to the ARM snippets. Maybe it belongs in a more general snippet collection?

File: VSCode/armsnippets.json [hunk 3]
```diff
@@ -34,6 +58,15 @@
     ],
     "description": "ARM Parameter"
 },
+"Deployment Parameter": {
+    "prefix": "arm-deplparameter",
```
Comment (philon-msft) [inline VSCode/armsnippets.json]: Rename to "arm-paramvalue"? Also update description to say "parameter value"

File: VSCode/armsnippets.json (hunk 1 above)
Comment (anderseide) [reply VSCode/armsnippets.json]: This snippet is intended for creating a new azuredeploy.parameters.json file, and adds the set of members recommended to have in that file. In difference from the one above, it also reference the correct schema (.../deploymentParameters.json vs. .../deploymentTemplate.json). Please correct me if I missunderstand your feedback

File: VSCode/armsnippets.json (hunk 2 above)
Comment (anderseide) [reply VSCode/armsnippets.json]: Will remove that one then

File: VSCode/armsnippets.json (hunk 3 above)
Comment (anderseide) [reply VSCode/armsnippets.json]: Way better name! Changing

File: VSCode/armsnippets.json (hunk 1 above)
Comment (philon-msft) [reply VSCode/armsnippets.json]: Sorry my mistake. Good idea!

**Pull request merged**